### PR TITLE
GPT-J CI

### PR DESCRIPTION
**Pull request by sunghyuckhong**
**##** 문제상황
- gptj ci test 정의 

## 목적(해결방향)
- 

## 구현 설명 Abstract
-gptj ci 정의 

## 구현 설명 Detail (ex. 추가된 argument)
- cnn_eval_accuracy_ci를 통해서 그래프 분리 기능 및 furiosa-llm이 정상작동하는걸 확인 (tokenwise로 combined graph + transformers == separated graphs + transformers == separated graphs + furiosa_llm_original 인 것을 확인) 

## test 통과 내역 (CI 통과내역과 어떤 machine (GPU pod or NPU machien or local)에서 테스트했는지)
- [ ] local machine with 3090
- [ ] GPU pod
- [ ] warboy pod
  - `"python main.py --scenario Offline --model-path ./model/ --dataset-path ./data/cnn_eval_accuracy_ci.json --model_script_path ./quantization/model_script/Qlevel4_RGDA0-W8A8KV8-PTQ-SMQ.yaml --gpu --accuracy --use_mcp --calib-dataset-path ./data/cnn_dailymail_calibration.json --recalibrate --model_source [transformers or furiosa_llm_original]"

## TODO (ex. 후속PR예고)
-

### CODE REVIEW

File: language/gpt-j/quantization/get_quant_model.py [hunk 1]
```diff
@@ -80,12 +80,12 @@ class dotdict(dict):
...
 
 
-def get_quant_model(model, calib_dataset_path, model_script_path, calib_without_padding, recalibrate):
+def get_quant_model(model, calib_dataset_path, model_script_path, calib_without_padding, recalibrate, qformat_path = None, qparam_path = None, immigrate_qparams = False):
     # Load model script and calibration dataloader (Refer to inference-compression/language/gpt-j/README.md on how to download evaluation and calibration dataset )
     model_script = load_model_script(model_script_path)
 
-    qformat_path = f"./quantization/output/qformat_{model_script_path.split('.')[1].split('/')[-1]}.yaml" 
-    qparam_path = f"./quantization/output/qparam_{model_script_path.split('.')[1].split('/')[-1]}.npy"
+    # qformat_path = f"./quantization/output/qformat_{model_script_path.split('.')[1].split('/')[-1]}.yaml"
```
Comment (BeomGeunCho) [inline language/gpt-j/quantization/get_quant_model.py]: qformat, qparam main.py 에서도 입력받도록 반영해야할 것 같은데 확인 부탁드립니다

File: language/gpt-j/main_ci.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    
+
+    torch_device_type = 'cuda' if torch.cuda.is_available() else 'cpu'
+    device = torch.device(torch_device_type)
+
+    
+    #for the sake of convenience 
+    from furiosa_llm_models.gptj.symbolic.huggingface_rope_rngd_gelu import GPTJForCausalLM
+    config = AutoConfig.from_pretrained("EleutherAI/gpt-j-6B")
+    config.num_hidden_layers = 4
```
Comment (jeongin-yun) [inline language/gpt-j/main_ci.py]: 전체 모델로 수정해주세요

File: language/gpt-j/main_ci.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    hf_model = GPTJForCausalLM.from_pretrained("EleutherAI/gpt-j-6B", config=config).to(device)
```
Comment (jeongin-yun) [inline language/gpt-j/main_ci.py]: hf_model 보단 golden 이라는 이름으로 변경해주세요

File: language/gpt-j/quantization/calibration_utils/make_calib_dataloader.py [hunk 4]
```diff
@@ -18,14 +18,14 @@ def make_calib_dataloader(calib_dataset_path, batch_size, calib_without_padding)
...
+            bucketized_input_ids[:, -starting_input_len:] = starting_input_ids
             
             starting_attention_mask =  data_object.source_encoded_attn_masks[idx]
             bucketized_attention_mask = torch.zeros((batch_size, bucket_size), dtype=torch.int)
-            bucketized_attention_mask[:, :starting_input_len] = starting_attention_mask
+            bucketized_attention_mask[:, -starting_input_len:] = starting_attention_mask
             
             starting_position_ids = torch.arange(len(data_object.source_encoded_input_ids[idx][0])).reshape(1,-1)
-            bucketized_position_ids = torch.cat([starting_position_ids, torch.zeros((1, bucket_size - starting_input_len), dtype=torch.long)], dim=1)
+            bucketized_position_ids = torch.cat([torch.zeros((1, bucket_size - starting_input_len), dtype=torch.long), starting_position_ids], dim=1)
```
Comment (jeongin-yun) [inline language/gpt-j/quantization/calibration_utils/make_calib_dataloader.py]: 이 부분은 golden 모델용으로 보이는데, 왜 수정됐나요?

Comment (sunghyuckhong) [reply language/gpt-j/quantization/calibration_utils/make_calib_dataloader.py]: 이전에 golden model vs mlperf submission qparam 동치를 확인하기 위해 left padding 으로 맞춰주는 작업을 하였는데, 같이 딸려 들어간 것 같습니다. change revert했습니다.

File: language/gpt-j/main_ci.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    # model = GPTJForCausalLM.from_pretrained(model_path, config=model_config)
+    
+
+    torch_device_type = 'cuda' if torch.cuda.is_available() else 'cpu'
+    device = torch.device(torch_device_type)
+
+    
+    #for the sake of convenience 
+    from furiosa_llm_models.gptj.symbolic.huggingface_rope_rngd_gelu import GPTJForCausalLM
+    config = AutoConfig.from_pretrained("EleutherAI/gpt-j-6B")
```
Comment (BeomGeunCho) [inline language/gpt-j/main_ci.py]: model_path 를 입력받아 mlperf gptj 로 생성되도록 변경 부탁드립니다

Comment (sunghyuckhong) [reply language/gpt-j/main_ci.py]: 반영했는데, 성환님 등 타팀에서도 이 test에 대한 수요가 있을 것 같아서 주석으로 hf model로 동치 확인하는 방법은 남겨두었습니다.

**Review by jeongin-yun — APPROVED** [review]
